app/controllers/assessments_controller.rb

```diff
@@ -5,83 +5,85 @@ class AssessmentsController < ApplicationController
   def index; end
 
   def new
-    # Don't bother with this if the submission token isn't the correct length
-    @patient_submission_token = params[:patient_submission_token].gsub(/[^0-9a-z]/i, '')
-    return if @patient_submission_token.length != 40
+    # Validate and get patient submission token and redirect if invalid link or already reported
+    @patient_submission_token = check_and_get_patient_submission_token(invalid_link_url,
+                                                                       ADMIN_OPTIONS['report_mode'] ? already_reported_report_url : already_reported_url)
+    return if @patient_submission_token.nil?
 
     # Don't bother with this if the jurisdiction unique identifier isn't at least 10 characters long
-    @unique_identifier = params[:unique_identifier]&.gsub(/[^0-9a-z]/i, '')
-    return if @unique_identifier.present? && @unique_identifier.length < 10
+    @unique_identifier = params.permit(:unique_identifier)[:unique_identifier]&.gsub(/[^0-9A-Za-z_-]/i, '')
```

Comment:
The `i` makes this regex already case insensitive so you shouldn't need to include the A-Z yeah?

Reply:
didn't know that, thanks @kylietmo !

Reply:
Full transparency - I always forget regex syntax so I just wanted to double check the regex and plopped it in here to be sure: https://regex101.com/r/r9f7cr/1

Reply:
Addressed here: https://github.com/SaraAlert/SaraAlert/pull/446/commits/38dc5f58019d8aacda9df4952d3b9152de1997fb